### DISCUSSION TITLE
Some issues about low-latency video stream

### DISCUSSION THREAD

**Discussion by OptimusTom426** [svpcom/wfb-ng#354] [Questions]
From the code point of view, it seems that the project achieves low latency by building Ethernet frames and sending video data as payload to LLC at the L2 layer to bypass kernel's TCP/IP protocol stack. Can other mechanisms (such as xdp/ebpf, dpdk, etc.) be used to  achieve the same effect?

**Comment by svpcom** [comment]
The main idea is bypass 802.11 (wifi) stack and use wifi card as raw radio. Classic wifi has very slow AP association and limited range due to mandatory ACKs. 

Wfb-ng use direct packet capture/injection via pcap and raw sockets. You can try dpdk but this will not give any acceleration, since the bottleneck is the speed of the radio channel

**Reply by OptimusTom426** [reply]
Is the latency also related to the CSMA/CA mechanism? How much influence does this mechanism have?

**Reply by svpcom** [reply]
All wfb-ng latency (including radio tx/rx) is less than 5ms. All other latency is from camera, video codecs and display.
